doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -159,104 +453,565 @@ for lossless profile based on the cable length and speed so that user can create
+            "type": "ingress",
+            "mode": "dynamic",
+            "xoff": "15728640"
+        },
+        "egress_lossless_pool": {
+            "size": "50665436",
+            "type": "egress",
+            "mode": "static"
+        }
+    },
+    "BUFFER_PROFILE": {
+        "ingress_lossy_profile": {
+            "pool":"[BUFFER_POOL|ingress_lossless_pool]",
+            "size":"0",
+            "static_th":"66394076"
+        },
+        "egress_lossless_profile": {
+            "pool":"[BUFFER_POOL|egress_lossless_pool]",
+            "size":"0",
+            "static_th":"67117468"
+        },
+        "egress_lossy_profile": {
+            "pool":"[BUFFER_POOL|egress_lossless_pool]",
+            "size":"0",
+            "mode": "dynamic",
+            "dynamic_th":"2"
+        }
+    },
+{%- endmacro %}
+
 ```
-### 2.4.4 Traffic Class to Queue mapping: (PFC Pause frames priority only 3 & 4 are supported HW)
+
+- Ingress is always single pool i.e "ingress_lossless_pool
+- With lossless configurations, pool XOFF size is reserved buffer for HDRM or RTT packets after generate pause.
+- Xoff should be configurable by user.
+- Egress also single pool to support lossless behavior in TH3 i.e  egress_lossless_pool
+- With this approach lossy traffic and lossless traffic will share buffers and better utilization of buffer for lossy traffic.
+- Pool size modifications are supported but pool delete and modifying pool names not supported.
+
+### 2.2.4 Moving to lossless buffers mode
+
+Change default buffers to lossless buffers to support lossless behavior. This changes requires reboot for clean initialization of MMU.
+
+- Lossless buffers will be initialized/de-initialized by "buffer init"/"roce enable".
+- Pool's for lossy and lossless are created traffic based on NPU.
+- Configure HDRM Pool in ingress pool
+- Apply PG3&PG4 with default lossless buffer profiles and PG7 with lossy profile
+- Apply Q3&Q4 with lossless buffer profiles and All other queues in system with lossy.
+
+"buffer init" supports only initializing only buffers at pool, PG and queue. All other required configurations to support ROCEv2 should be done by user.
+
+To minimize user configurations mistakes this release will enhanced to provide "roce enable" to set ROCEv2 default configurations.
+
+### 2.2.5 Moving from lossless mode to defaults
+
+SONiC clears buffer configuration from NOS to bring it to original defaults by "buffer deinit"/"roce disable". This changes requires reboot for clean initialization of MMU.
+
+"buffer de-init" supports clearing only buffers at pool, PG and queue. All other required configurations should be removed by user to disable ROCEv2.
+
+To minimize user configurations mistakes this release will enhanced clear all roce defaults by "roce disable".
+
+### 2.2.6 Dynamic buffer configuration change
+
+NPU has limitations for changing buffer configurations when traffic is running on PG/Q or port, which leads to MMU to corrupt.

```

Comment:
1#No way to know to user, unless we see some traffic loss.                                                                                                      
2# MMU corrupted because of NOS trigger configuration which uses SDK API. it will come to functionality issue.                                                  3# e-sonic, decided to go with safer approach to avoid corruption, Which is approved by PLM

Reply:
thanks for the clarification